File: 01-csharp/demo-01/UnitTestingAndTDD/Sample02/ExternalFizzBuzz.cs
using Sample01;
using Sample02.Abstractions;

namespace Sample02
{
    public class ExternalFizzBuzz : FizzBuzz
    {
        private readonly IFizzBuzzService service;
        public ExternalFizzBuzz(IFizzBuzzService service) :base()
        {
            this.service=service;    
        }

        public override string MultipleOf3(int number)
        {
            return service.MultipleOf3(number);
        }

        public override string MultipleOf5(int number)
        {
            return service.MultipleOf5(number);
        }

        public override string MultipleOf3And5(int number)
        {
            return service.MultipleOf3And5(number);
        }
    }
}

File: 02-java/demo-01/UnitTestingAndTDD/Sample02/src/main/java/sample02/ExternalFizzBuzz.java
package sample02;
import sample01.FizzBuzz;
import sample02.abstractions.IFizzBuzzService;

public class ExternalFizzBuzz extends FizzBuzz {
    private final IFizzBuzzService service;
    public ExternalFizzBuzz(IFizzBuzzService service) {
        super();
        this.service=service;
    }

    @Override
    public String multipleOf3(int number) {
        return service.multipleOf3(number);
    }

    @Override
    public String multipleOf5(int number) {
        return service.multipleOf5(number);
    }

    @Override
    public String multipleOf3And5(int number) {
        return service.multipleOf3And5(number);
    }
}
